diff --git a/adapter/provider/adapter.go b/adapter/provider/adapter.go
index 8b1e75e45..b7f3f20ff 100644
--- a/adapter/provider/adapter.go
+++ b/adapter/provider/adapter.go
@@ -118,9 +118,9 @@ func (a *Adapter) resolveOutboundTags(newOpts []option.Outbound) []string {
 	for i, opt := range newOpts {
 		var baseTag string
 		if opt.Tag != "" {
-			baseTag = F.ToString(a.providerTag, "/", opt.Tag)
+			baseTag = opt.Tag
 		} else {
-			baseTag = F.ToString(a.providerTag, "/", i)
+			baseTag = F.ToString(i)
 		}
 		tag := baseTag
 		for n := 2; seen[tag]; n++ {
@@ -302,13 +302,13 @@ func (a *Adapter) RewriteDetourForProvider(opts []option.Outbound, endpointOpts
 	tagMapping := make(map[string]string)
 	for _, opt := range opts {
 		if opt.Tag != "" {
-			tagMapping[opt.Tag] = F.ToString(a.providerTag, "/", opt.Tag)
+			tagMapping[opt.Tag] = opt.Tag
 		}
 	}
 	for _, endpoints := range endpointOpts {
 		for _, opt := range endpoints {
 			if opt.Tag != "" {
-				tagMapping[opt.Tag] = F.ToString(a.providerTag, "/", opt.Tag)
+				tagMapping[opt.Tag] = opt.Tag
 			}
 		}
 	}
@@ -327,13 +327,13 @@ func (a *Adapter) RewriteDetourForProviderEndpoints(opts []option.Endpoint, outb
 	tagMapping := make(map[string]string)
 	for _, opt := range opts {
 		if opt.Tag != "" {
-			tagMapping[opt.Tag] = F.ToString(a.providerTag, "/", opt.Tag)
+			tagMapping[opt.Tag] = opt.Tag
 		}
 	}
 	for _, outbounds := range outboundOpts {
 		for _, opt := range outbounds {
 			if opt.Tag != "" {
-				tagMapping[opt.Tag] = F.ToString(a.providerTag, "/", opt.Tag)
+				tagMapping[opt.Tag] = opt.Tag
 			}
 		}
 	}
@@ -354,9 +354,9 @@ func (a *Adapter) resolveEndpointTags(newOpts []option.Endpoint) []string {
 	for i, opt := range newOpts {
 		var baseTag string
 		if opt.Tag != "" {
-			baseTag = F.ToString(a.providerTag, "/", opt.Tag)
+			baseTag = opt.Tag
 		} else {
-			baseTag = F.ToString(a.providerTag, "/endpoint-", i)
+			baseTag = F.ToString("endpoint-", i)
 		}
 		tag := baseTag
 		for n := 2; seen[tag]; n++ {
